File: app/src/main/java/hu/bme/szn/randomjokes/RandomJokesApplicationComponent.java
package hu.bme.szn.randomjokes;
import javax.inject.Singleton;

import dagger.Component;
import hu.bme.szn.randomjokes.network.NetworkModule;
import hu.bme.szn.randomjokes.ui.UIModule;
import hu.bme.szn.randomjokes.interactor.jokes.JokesInteractor;
import hu.bme.szn.randomjokes.ui.jokes.JokeActivity;
import hu.bme.szn.randomjokes.ui.jokes.JokePresenter;

@Singleton
@Component(modules = {UIModule.class, NetworkModule.class})
public interface RandomJokesApplicationComponent {
    void inject(JokeActivity jokeActivity);

    void inject(JokesInteractor jokesInteractor);

    void inject(JokePresenter jokesPresenter);
}

File: app/src/main/java/hu/bme/szn/randomjokes/RandomJokesApplicationComponent.kt
package hu.bme.szn.randomjokes

import dagger.Component
import hu.bme.szn.randomjokes.interactor.InteractorModule
import hu.bme.szn.randomjokes.network.NetworkModule
import hu.bme.szn.randomjokes.ui.UIModule
import hu.bme.szn.randomjokes.ui.about.AboutActivity
import hu.bme.szn.randomjokes.ui.jokes.JokeActivity
import javax.inject.Singleton

@Singleton
@Component(modules = [UIModule::class, NetworkModule::class, InteractorModule::class])
interface RandomJokesApplicationComponent {
    fun inject(jokeActivity: JokeActivity)
    fun inject(aboutActivity: AboutActivity)
}